script.js

```diff
@@ -1,12 +1,120 @@
-//You can edit ALL of the code here
+
 function setup() {
   const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
+  console.log(allEpisodes);
 }
 
+
+   //This is header-Displaying Number Code 
 function makePageForEpisodes(episodeList) {
+  
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  rootElem.innerHTML = "";
+  addSelectFunctionality(episodeList);
+  addSearchFunctionality(episodeList);
+  const episodeCountElem = document.createElement("p");
+  episodeCountElem.className = "episode-count";
+  episodeCountElem.textContent = `Displaying ${episodeList.length} / ${getAllEpisodes().length} episodes`;
+  rootElem.appendChild(episodeCountElem);
+
+  //This big container for the Movie(grid-container)
+  const gridContainer = document.createElement("div");
+  gridContainer.className = "grid-container";
+  rootElem.appendChild(gridContainer);
+
+  //This is the each movie Card container Code(grid-item)
+  episodeList.forEach((episode) => {
+    const gridItem = document.createElement("div");
+    gridItem.className = "grid-item";
+    gridContainer.appendChild(gridItem);
+
+   //This is Movie Title Code both episode-Name and episode-Code
+    const episodeCode = getEpisodeCode(episode);
+    const episodeTitle = `${episode.name} - ${episodeCode} `;
+    const episodeTitleElem = document.createElement("h2");
+    episodeTitleElem.textContent = episodeTitle;
+    gridItem.appendChild(episodeTitleElem);
+   
+    //This is Image Code
+    const imgElem = document.createElement("img");
+    imgElem.src = episode.image.medium;
+    imgElem.alt = episodeTitle;
```

Comment:
Nice and welcome addition for accessibility!

Reply:
Thank you very much for your time.